### PR TITLE
Good day plex!

### PR DESCRIPTION
**Pull request by Hellowlol**
This pr will get all the server connections that you own and are online and add that to a dropdown in the welcome page. If you change server it will enable/change your settings according to what it found on plex.tv.

### CODE REVIEW

**Comment by Hellowlol** [comment]
rebased to dev

File: data/interfaces/default/welcome.html [hunk 1]
```diff
@@ -399,7 +470,19 @@ <h3 style="line-height: 50px;">Welcome!</h3>
             }
         });
     });
+
+    function getServerOptions(token) {
+        /* Set token and returns server options */
+        $.ajax({
+            url: "discover/" + token,
```
Comment (drzoidberg33) [inline data/interfaces/default/welcome.html:477]: Curious why you just didn't do this all in javascript and access plex.tv directly?

Comment (Hellowlol) [inline data/interfaces/default/welcome.html:477]: I dont like js :P, and if i did i cant use it in the api later.

Comment (drzoidberg33) [inline data/interfaces/default/welcome.html:477]: Okay, no problem. Will test and provide feedback.

**Comment by drzoidberg33** [comment]
Does this work on your end? I'm just getting this on the `/discover` endpoint:

```
500 Internal Server Error

The server encountered an unexpected condition which prevented it from fulfilling the request.
Traceback (most recent call last):
  File "/Users/tim/PyCharmProjects/plexpy-hl2/lib/cherrypy/_cprequest.py", line 670, in respond
    response.body = self.handler()
  File "/Users/tim/PyCharmProjects/plexpy-hl2/lib/cherrypy/lib/encoding.py", line 217, in __call__
    self.body = self.oldhandler(*args, **kwargs)
  File "/Users/tim/PyCharmProjects/plexpy-hl2/lib/cherrypy/_cpdispatch.py", line 61, in __call__
    return self.callable(*self.args, **self.kwargs)
  File "/Users/tim/PyCharmProjects/plexpy-hl2/plexpy/webserve.py", line 1457, in discover
    servers = result.discover()
  File "/Users/tim/PyCharmProjects/plexpy-hl2/plexpy/plextv.py", line 421, in discover
    s.update(server)
AttributeError: 'unicode' object has no attribute 'update'
```

**Comment by Hellowlol** [comment]
s should be a dict not unicode. It worked when i sent the pr. I have to test when i get home

**Comment by Hellowlol** [comment]
Can you manually try this:

```
import xmltodict
import json

x = """<?xml version="1.0" encoding="UTF-8"?>
<MediaContainer size="2">
  <Device name="server 1" product="Plex Media Server" productVersion="0.9.12.11.1406-8403350" platform="Windows" platformVersion="6.1 (Build 7601)" device="PC" clientIdentifier="1324" createdAt="1382901845" lastSeenAt="1443531307" provides="server" owned="1" httpsRequired="0" synced="0" publicAddressMatches="0" presence="1">
    <Connection protocol="https" address="10.0.0.97" port="32400" uri="x" local="1"/>
    <Connection protocol="https" address="y.y.y.y.y.y." port="19953" uri="y.y.y.y.y" local="0"/>
  </Device>
  <Device name="test-laptop" product="Plex Media Server" productVersion="0.9.12.11.1406-8403350" platform="Windows" platformVersion="6.1 (Build 7601)" device="PC" clientIdentifier="134" createdAt="1443297872" lastSeenAt="1443479540" provides="server" owned="1" httpsRequired="0" synced="0" publicAddressMatches="0" presence="0">
    <Connection protocol="https" address="10.0.0.53" port="32400" uri="x" local="1"/>
    <Connection protocol="https" address="x.x.x.x.x" port="17086" uri="x.x.x.x.x.x.x.x" local="0"/>
  </Device>
</MediaContainer>"""


clean_servers = []
servers = xmltodict.parse(x, attr_prefix='')

for server in servers['MediaContainer']['Device']:
    if server.get('presence', None) == '1' and server.get('owned', None) == '1':
        for s in server['Connection']:
            s.update(server)
            s['value'] = s['address']
            s['label'] = server['name']
            del s['Connection']
            clean_servers.append(s)

print json.dumps(clean_servers, indent=4)
```

And see if you get any error

**Comment by JonnyWong16** [comment]
Discover works on my end.

It still needs the ability to manually enter an IP address. Use an `input` instead of `select`.
Styling needs to be fixed (box changes size on focus/blur).
It should indicate which IP addresses are local/remote in the list.

**Comment by Hellowlol** [comment]
@JonnyWong16 
It support manually entering ips, just press enter and start writing. I dont think remote ips need to made more visual. It's easy to see the difference and by choosing a remote ip the remote check box is automatically checked. Same goes for ssl if ssl is required.

I agree on focus/blur this needs to be addressed but I'm really awefull at it so I hoped you guys could fix that

**Comment by drzoidberg33** [comment]
Try with this:

```
<?xml version="1.0" encoding="UTF-8"?>
               <MediaContainer size="4">
                  <Device name="xyz" product="Plex Media Server" productVersion="xyz" platform="Linux" platformVersion="2.6.32-26-pve (#1 SMP Mon Oct 14 08:22:20 CEST 2013)" device="PC" clientIdentifier="xyz" createdAt="1423260839" lastSeenAt="1443518582" provides="server" owned="0" httpsRequired="0" ownerId="123" home="1" accessToken="xyz" sourceTitle="drzoidberg33" synced="0" publicAddressMatches="0" presence="1">
                    <Connection protocol="https" address="192.168.1.10" port="32400" uri="https://xyp.plex.direct:32400" local="1"/>
                    <Connection protocol="https" address="1.2.3.4" port="8088" uri="https://xyz.plex.direct:8088" local="0"/>
                  </Device>
                  <Device product="Plex Media Server" productVersion="0.9.7.12" platform="" platformVersion="" device="" clientIdentifier="xyz" createdAt="1439067530" lastSeenAt="1443513242" owned="0" provides="server" ownerId="123" home="1" accessToken="xyz" sourceTitle="drzoidberg33" name="Cloud Sync" synced="1" publicAddressMatches="0" presence="0">
                    <Connection protocol="https" address="plex.tv" port="443" uri="https://plex.tv" local="0"/>
                  </Device>
                  <Device name="Plex-HTPC" product="Plex Media Server" productVersion="xyz" platform="Linux" platformVersion="2.6.32-504.30.3.el6.x86_64 (#1 SMP Wed Jul 15 10:13:09 UTC 2015)" device="PC" clientIdentifier="xyz" createdAt="1388764303" lastSeenAt="1443518774" provides="server" owned="0" httpsRequired="0" ownerId="123" home="1" accessToken="xyz" sourceTitle="drzoidberg33" synced="0" publicAddressMatches="0" presence="1">
                    <Connection protocol="https" address="192.168.1.3" port="32400" uri="https://xyz.plex.direct:32400" local="1"/>
                    <Connection protocol="https" address="2.3.4.5" port="32400" uri="https://xyz.plex.direct:32400" local="0"/>
                  </Device>
                  <Device name="My LG D802" product="Plex for Android" productVersion="4.10.2.424" platform="Android" platformVersion="5.0.2" device="LG-D802" clientIdentifier="xyz-com-plexapp-android" createdAt="1439067425" lastSeenAt="1443469138" provides="player,controller,sync-target" owned="1" accessToken="xyz" publicAddressMatches="0" presence="0">
                    <Connection protocol="http" address="192.168.1.11" port="32400" uri="http://192.168.1.11:32400" local="1"/>
                  </Device>
               </MediaContainer>
```

**Comment by Hellowlol** [comment]
@drzoidberg33 It works as it should here, it returns a empty list as it should (none of the servers listed is "owned" why you get that error i dont know, it should never get that far.)

**Comment by drzoidberg33** [comment]
Sorry, my one server wasn't listed when I ran that.

The problem is the loop (`for s in server['Connection']:`) - the code is assuming that there will be multiple dicts here, but there aren't always like below:

```
<Device name="MacBook Server" product="Plex Media Server" productVersion="" platform="MacOSX" platformVersion="10.11.0" device="Mac" clientIdentifier="" createdAt="1421915745" lastSeenAt="1443547212" provides="server" owned="1" accessToken="" httpsRequired="0" synced="0" publicAddressMatches="1" presence="1">
    <Connection protocol="https" address="192.168.1.10" port="32400" uri="https://192-168-1-10.plex.direct:32400" local="1"/>
  </Device>
```

Now instead of it looping through each `Connection` dict, it just loops through the keys in the one dict.

**Comment by Hellowlol** [comment]
Have you tested the pr after my last commit? it should be ok now.

**Comment by drzoidberg33** [comment]
Sorry haven't had time yet, will test tonight.

**Comment by drzoidberg33** [comment]
Doesn't seem to work. It just sets the server to 127.0.0.1 no matter what server I choose.

The list now works, but when completing the setup wizard it reverts to 127.0.0.1.

**Comment by Hellowlol** [comment]
Works fine here. WTF. Where do you see that it reverts to 127.0.0.1 ? Do you change any other settings in the wizard except from the dropdown?

**Comment by drzoidberg33** [comment]
Nope, just tried again. I select a remote server, finish the wizard and then if I check the settings the server IP is 127.0.0.1 while the port and other settings seem to be correct.

**Comment by Hellowlol** [comment]
Ah,crap i found the error, the ip isnt serialized, so when updateConfig is posted it just reverts to the default value. On a clean config thats 127.0.0.1, i didnt catch it as i never used a clean config.

**Comment by Hellowlol** [comment]
Works now :P

**Comment by Hellowlol** [comment]
Any chance you can test this again?

**Comment by drzoidberg33** [comment]
Works now, thanks!

**Pull request merged**